### DISCUSSION TITLE
SSL Provider status shows as "Disabled

### DISCUSSION THREAD

**Discussion by fandd1234** [requarks/wiki#4692] [Help / Questions]
Hi all,
I'm trying to enable the SSL function in the config.yml and after change the value to "True", it still doesn't work.
![1234](https://user-images.githubusercontent.com/92530680/141452192-33beeeb6-6e75-4c7a-b788-ea45d42291a4.JPG)

When I check the SSL provider status, it shows as "Disabled" as the screenshot attached.
May I ask if there's anything I missed to configure on it?  Please advise, many thanks.

**Comment by ieaster1** [comment]
Have you restarted the node server after the config change?

**Comment by fandd1234** [comment]
@ieaster1 I've restarted the node server by using "node server restart" and turn out the error as below.
error: mac verify failure, is it something wrong with my cert file setup ?
![wiki_1](https://user-images.githubusercontent.com/92530680/145185160-824fc90a-3969-4a38-aba4-e2d8926b0b3f.JPG)

**Reply by Phil-hostwiki** [reply]
Have you considered using NGINX reverse proxy and Letsencrypt?

**Reply by fandd1234** [reply]
@Phil-hostwiki not yet, as I have my own pfx cert file and think that this should be a much more easier way to do that..
for Letsencrypt, I remember need to rearm the cert expiry date for 3 months, right? and I do not know much about the usage of NGINX reverse proxy...  thanks. :)

**Reply by Phil-hostwiki** [reply]
I see you already have your own cert.

With NGINX reverse proxy and let's encrypt, certbot takes care of SSL renewals automatically. So you don't have to do anything else after setup.

**Reply by fandd1234** [reply]
I will consider to use let's encrypt after my final try out on the current setup.  Actually I just managed to able to see my cert from my wiki page (as below screenshot).  But when I try to change the port to 443 in the config.yml, the page will gone down even I restart the node server.  Any clues why was that happen?


![SSL1](https://user-images.githubusercontent.com/92530680/145219074-4def62b6-f6d5-4c4c-924b-c9f560d54dcb.JPG)

**Comment by ieaster1** [comment]
Best guess is that your pfx cert is encrypted with a passphrase.  You would need to add that encryption password to the `passphrase` option rather than having it set to `null`.

You can also check the cert to make sure there is no issues with the file:
`openssl pkcs12 -in /path/to/your/cert.pfx -info -nokeys`

**Reply by fandd1234** [reply]
yes - I did put the passphrase back into the config.yml, but once I define to use port 443, I still couldn't manage to get that in from browser... is there any specific log that I can look into for finding out the problems?
